### PR TITLE
Add tests for observer and all uncovered methods.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/test/java/io/opentelemetry/metrics/CounterDoubleTest.java [hunk 1]
```diff
@@ -70,6 +70,13 @@ public void preventNull_Unit() {
     meter.counterDoubleBuilder("metric").setUnit(null).build();
   }
 
+  @Test
+  public void preventNull_LabelKeys() {
```
Comment (songy23) [inline api/src/test/java/io/opentelemetry/metrics/CounterDoubleTest.java]: For better coverage, consider adding a test `preventNull_LabelKey`, i.e `setLabelKeys(Arrays.<String>asList(null))`. Same for other tests.

File: api/src/test/java/io/opentelemetry/metrics/ObserverDoubleTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  private final Meter meter = OpenTelemetry.getMeter();
+
+  @Test
+  public void preventNonPrintableName() {
+    thrown.expect(IllegalArgumentException.class);
+    meter.observerDoubleBuilder("\2").build();
+  }
+
+  @Test
+  public void preventTooDoubleMeasureName() {
```
Comment (songy23) [inline api/src/test/java/io/opentelemetry/metrics/ObserverDoubleTest.java]:
```suggestion
  public void preventTooLongMeasureName() {
```

Comment (bogdandrutu) [reply api/src/test/java/io/opentelemetry/metrics/ObserverDoubleTest.java]: Done everywhere.

File: api/src/test/java/io/opentelemetry/metrics/CounterDoubleTest.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/test/java/io/opentelemetry/metrics/CounterDoubleTest.java]: Done.

**Pull request merged**